tap_github/streams.py

```diff
@@ -0,0 +1,753 @@
+from datetime import datetime
+import singer
+from singer import (metrics, bookmarks, metadata)
+
+LOGGER = singer.get_logger()
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+def get_bookmark(state, repo, stream_name, bookmark_key, start_date):
+    """
+    Return bookmark value if available in the state otherwise return start date
+    """
+    repo_stream_dict = bookmarks.get_bookmark(state, repo, stream_name)
+    if repo_stream_dict:
+        return repo_stream_dict.get(bookmark_key)
+
+    return start_date
+
+def get_schema(catalog, stream_id):
+    """
+    Return catalog of the specified stream.
+    """
+    stream_catalog = [cat for cat in catalog if cat['tap_stream_id'] == stream_id ][0]
+    return stream_catalog
+
+def get_child_full_url(child_object, repo_path, parent_id, grand_parent_id):
+    """
+    Build the child stream's URL based on the parent and the grandparent's ids.
+    """
+
+    if child_object.is_repository:
+        # The `is_repository` represents that the url contains /repos and the repository name.
+        child_full_url = '{}/repos/{}/{}'.format(
+            child_object.url,
+            repo_path,
+            child_object.path).format(*parent_id)
+
+    elif child_object.is_organization:
+        # The `is_organization` represents that the url contains the organization name.
+        org = repo_path.split('/')[0]
+        child_full_url = '{}/{}'.format(
+            child_object.url,
+            child_object.path).format(org, *parent_id, *grand_parent_id)
+
+    else:
```

Comment:
Is that mean else part is called without repo or organization? can you please add a code comment here?

Reply:
Yes, else part is called without repo or organization. Added comment in the code.